File: fix_result_save.py
# -*- coding: utf-8 -*-
# 修复result.html中的图片保存问题

with open('c:/tmp/emotion-share/result.html', 'r', encoding='utf-8') as f:
    content = f.read()

# 找到要替换的代码段
old_code = """            try {
                // 显示加载状态
                button.disabled = true;
                button.textContent = '生成中...';

                // 获取要截图的元素
                const element = document.getElementById('app');

                // 配置选项
                const options = {
                    backgroundColor: '#ffffff', // 纯白背景
                    scale: 2, // 提高清晰度
                    useCORS: true, // 允许跨域图片
                    logging: false,
                    width: element.scrollWidth,
                    height: element.scrollHeight,
                    windowWidth: element.scrollWidth,
                    windowHeight: element.scrollHeight
                };

                // 生成截图
                const canvas = await html2canvas(element, options);

                // 转换为blob并下载
                canvas.toBlob((blob) => {
                    const url = URL.createObjectURL(blob);
                    const link = document.createElement('a');
                    link.download = `情绪颗粒度测试结果_${new Date().getTime()}.png`;
                    link.href = url;
                    link.click();

                    // 清理
                    URL.revokeObjectURL(url);

                    // 恢复按钮状态
                    button.disabled = false;
                    button.innerHTML = '💾 保存图片';
                }, 'image/png', 1.0);

            } catch (error) {
                console.error('截图失败:', error);
                alert('截图失败，请重试或使用长按截图方式保存');

                // 恢复按钮状态
                button.disabled = false;
                button.innerHTML = '💾 保存图片';
            }"""

new_code = """            try {
                // 显示加载状态
                button.disabled = true;
                button.textContent = '生成中...';

                // 获取要截图的元素
                const element = document.getElementById('app');

                // 隐藏波浪装饰，避免截图时出现白色覆盖层
                const waveDecoration = document.querySelector('.wave-decoration');
                const originalWaveDisplay = waveDecoration ? waveDecoration.style.display : '';
                if (waveDecoration) {
                    waveDecoration.style.display = 'none';
                }

                // 给截图元素临时添加背景，确保截图清晰
                const originalBg = element.style.backgroundColor;
                element.style.backgroundColor = '#F8F9FA';

                // 配置选项
                const options = {
                    backgroundColor: '#F8F9FA', // 浅灰背景，与页面一致
                    scale: 2, // 提高清晰度
                    useCORS: true, // 允许跨域图片
                    allowTaint: true, // 允许污染的canvas
                    logging: false,
                    width: element.scrollWidth,
                    height: element.scrollHeight,
                    windowWidth: element.scrollWidth,
                    windowHeight: element.scrollHeight,
                    // 忽略波浪装饰元素
                    ignoreElements: (el) => {
                        return el.classList && el.classList.contains('wave-decoration');
                    }
                };

                // 生成截图
                const canvas = await html2canvas(element, options);

                // 恢复原始样式
                if (waveDecoration) {
                    waveDecoration.style.display = originalWaveDisplay;
                }
                element.style.backgroundColor = originalBg;

                // 转换为blob并下载
                canvas.toBlob((blob) => {
                    const url = URL.createObjectURL(blob);
                    const link = document.createElement('a');
                    link.download = `情绪颗粒度测试结果_${new Date().getTime()}.png`;
                    link.href = url;
                    link.click();

                    // 清理
                    URL.revokeObjectURL(url);

                    // 恢复按钮状态
                    button.disabled = false;
                    button.innerHTML = '💾 保存图片';
                }, 'image/png', 1.0);

            } catch (error) {
                console.error('截图失败:', error);
                alert('截图失败，请重试或使用长按截图方式保存');

                // 恢复按钮状态
                button.disabled = false;
                button.innerHTML = '💾 保存图片';

                // 确保恢复原始样式
                const waveDecoration = document.querySelector('.wave-decoration');
                if (waveDecoration) {
                    waveDecoration.style.display = '';
                }
                const element = document.getElementById('app');
                if (element) {
                    element.style.backgroundColor = '';
                }
            }"""

# 替换
new_content = content.replace(old_code, new_code)

if new_content == content:
    print("错误：未找到要替换的代码")
else:
    # 写入文件
    with open('c:/tmp/emotion-share/result.html', 'w', encoding='utf-8') as f:
        f.write(new_content)
    print("✅ 图片保存问题已修复！")
    print("修复内容：")
    print("1. 截图前隐藏波浪装饰元素")
    print("2. 添加浅灰色背景（#F8F9FA）替代纯白背景")
    print("3. 截图后恢复原始样式")
    print("4. 添加 ignoreElements 配置忽略波浪装饰")

File: fix_result_save.js
// 修复result.html中保存图片的白色覆盖问题
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'result.html');
let content = fs.readFileSync(filePath, 'utf8');

// 查找并替换saveImage函数
const oldCode = `            try {
                // 显示加载状态
                button.disabled = true;
                button.textContent = '生成中...';

                // 获取要截图的元素
                const element = document.getElementById('app');

                // 隐藏波浪装饰，避免截图时出现白色覆盖层
                const waveDecoration = document.querySelector('.wave-decoration');
                const originalWaveDisplay = waveDecoration ? waveDecoration.style.display : '';
                if (waveDecoration) {
                    waveDecoration.style.display = 'none';
                }

                // 给截图元素临时添加背景，确保截图清晰
                const originalBg = element.style.backgroundColor;
                element.style.backgroundColor = '#F8F9FA';`;

const newCode = `            try {
                // 显示加载状态
                button.disabled = true;
                button.textContent = '生成中...';

                // 获取要截图的元素
                const element = document.getElementById('app');
                const resultContent = document.querySelector('.result-content');

                // 隐藏波浪装饰，避免截图时出现白色覆盖层
                const waveDecoration = document.querySelector('.wave-decoration');
                const originalWaveDisplay = waveDecoration ? waveDecoration.style.display : '';
                if (waveDecoration) {
                    waveDecoration.style.display = 'none';
                }

                // 移除result-content的padding-top，避免顶部空白
                const originalPaddingTop = resultContent ? resultContent.style.paddingTop : '';
                if (resultContent) {
                    resultContent.style.paddingTop = '0';
                }

                // 给截图元素临时添加背景和内边距，确保截图美观
                const originalBg = element.style.backgroundColor;
                const originalPadding = element.style.padding;
                element.style.backgroundColor = '#F8F9FA';
                element.style.padding = '32px 24px';`;

// 替换第一部分
content = content.replace(oldCode, newCode);

// 替换恢复样式的部分
const oldRestore = `                // 恢复原始样式
                if (waveDecoration) {
                    waveDecoration.style.display = originalWaveDisplay;
                }
                element.style.backgroundColor = originalBg;`;

const newRestore = `                // 恢复原始样式
                if (waveDecoration) {
                    waveDecoration.style.display = originalWaveDisplay;
                }
                if (resultContent) {
                    resultContent.style.paddingTop = originalPaddingTop;
                }
                element.style.backgroundColor = originalBg;
                element.style.padding = originalPadding;`;

content = content.replace(oldRestore, newRestore);

// 替换错误处理中的恢复部分
const oldErrorRestore = `                // 确保恢复原始样式
                const waveDecoration = document.querySelector('.wave-decoration');
                if (waveDecoration) {
                    waveDecoration.style.display = '';
                }
                const element = document.getElementById('app');
                if (element) {
                    element.style.backgroundColor = '';
                }`;

const newErrorRestore = `                // 确保恢复原始样式
                const waveDecoration = document.querySelector('.wave-decoration');
                if (waveDecoration) {
                    waveDecoration.style.display = '';
                }
                const resultContent = document.querySelector('.result-content');
                if (resultContent) {
                    resultContent.style.paddingTop = '';
                }
                const element = document.getElementById('app');
                if (element) {
                    element.style.backgroundColor = '';
                    element.style.padding = '';
                }`;

content = content.replace(oldErrorRestore, newErrorRestore);

// 写回文件
fs.writeFileSync(filePath, content, 'utf8');
console.log('✅ 修复完成！已解决保存图片时的白色覆盖问题。');
